collectors/logsmanagement.plugin/plugin_logsmanagement_web_log.c

```diff
@@ -0,0 +1,995 @@
+                , "timings"
+                , NULL
+                , "Request Processing Time"
+                , "milliseconds"
+                , "logsmanagement.plugin"
+                , NULL
+                , NETDATA_CHART_PRIO_REQ_PROC_TIME
+                , p_file_info->update_every
+                , RRDSET_TYPE_LINE
+        );
+        chart_data->dim_req_proc_time_min = rrddim_add(chart_data->st_req_proc_time, "min", NULL, 1, 1000, RRD_ALGORITHM_ABSOLUTE);
+        chart_data->dim_req_proc_time_max = rrddim_add(chart_data->st_req_proc_time, "max", NULL, 1, 1000, RRD_ALGORITHM_ABSOLUTE);
+        chart_data->dim_req_proc_time_avg = rrddim_add(chart_data->st_req_proc_time, "avg", NULL, 1, 1000, RRD_ALGORITHM_ABSOLUTE);
+    }
+
+    /* Response code family - initialise */
+    if(p_file_info->parser_config->chart_config & CHART_RESP_CODE_FAMILY){
+        chart_data->st_resp_code_family = rrdset_create_localhost(
+                (char *) p_file_info->chart_name
+                , "responses"
+                , NULL
+                , "responses"
+                , NULL
+                , "Response Codes"
+                , "requests/s"
+                , "logsmanagement.plugin"
+                , NULL
+                , NETDATA_CHART_PRIO_RESP_CODE_FAMILY
+                , p_file_info->update_every
+                , RRDSET_TYPE_AREA
+        );
+        chart_data->dim_resp_code_family_1xx = rrddim_add(chart_data->st_resp_code_family, "1xx", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_resp_code_family_2xx = rrddim_add(chart_data->st_resp_code_family, "2xx", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_resp_code_family_3xx = rrddim_add(chart_data->st_resp_code_family, "3xx", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_resp_code_family_4xx = rrddim_add(chart_data->st_resp_code_family, "4xx", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_resp_code_family_5xx = rrddim_add(chart_data->st_resp_code_family, "5xx", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_resp_code_family_other = rrddim_add(chart_data->st_resp_code_family, "other", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);     
+    }   
+
+    /* Response code - initialise */
+    if(p_file_info->parser_config->chart_config & CHART_RESP_CODE){
+        chart_data->st_resp_code = rrdset_create_localhost(
+                (char *) p_file_info->chart_name
+                , "detailed responses"
+                , NULL
+                , "responses"
+                , NULL
+                , "Detailed Response Codes"
+                , "requests/s"
+                , "logsmanagement.plugin"
+                , NULL
+                , NETDATA_CHART_PRIO_RESP_CODE
+                , p_file_info->update_every
+                , RRDSET_TYPE_AREA
+        );
+        for(int j = 0; j < 500; j++){
+            char dim_resp_code_name[4];
+            sprintf(dim_resp_code_name, "%d", j + 100);
```

Comment:
Like previous, I suggest to use `snprintfz`. 
Well, 500 dimensions on a chart looks like something unnecessary initially. Why do not you create a chart with the expected dimensions and create them when there is a real request?

Reply:
This is a good point, I shall update this not only for this chart, but others too. I usually create all the possible dimensions in charts with a fixed maximum number, but that's wasted resources.

Reply:
@thiagoftsm That's now changed to only create the "other" dimension by default. All the other dimensions are created once there is a request. However, the problem is that this way, we miss the first collection, because the algorithm is `RRD_ALGORITHM_INCREMENTAL` and there is no previous value to be used when the dimension is created.

Any ideas how to solve this?

Reply:
Unless I am missing something, when we do not have any previous value, it means previous value was zero. :thinking:

Reply:
Apparently that's the way it's meant to work for any dimensions where `RRD_ALGORITHM_INCREMENTAL`, so I am marking this one as resolved.